purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -70,19 +121,40 @@ def _inter_company_create_sale_order(self, dest_company):
         # Accessing to selling partner with selling user, so data like
         # property_account_position can be retrieved
         company_partner = self.company_id.partner_id
-        # check pricelist currency should be same with PO/SO document
-        if self.currency_id.id != (
-            company_partner.property_product_pricelist.currency_id.id
+        intercompany_so_pricelist = company_partner.property_product_pricelist
+        # pricelist currency should be same with PO/SO document
+        # if corresponding pricelist currency is not the same
+        # find or create propriate pricelist
+        if intercompany_so_pricelist and (
+            self.currency_id == intercompany_so_pricelist.currency_id
         ):
-            raise UserError(
-                _(
-                    "You cannot create SO from PO because "
-                    "sale price list currency is different than "
-                    "purchase price list currency."
+            related_pricelist = intercompany_so_pricelist
+        else:
+            related_pricelist = pricelist_obj.search(
+                [
+                    ("currency_id", "=", self.currency_id.id),
+                    ("company_id", "=", dest_company.id),
+                ]
+            )
+            if len(related_pricelist) > 1:
+                raise UserError(
+                    _(
+                        "There is more than one pricelist for this vendor "
+                        "with the same currency"
+                    )
                 )
```

Comment:
Why do you raise this error instead of just fetching 1 of the many pricelists?

Reply:
customer request was to have only one pricelist for vendor company with the same currency
it's a signal to check and remove not needed one